packages/payment-detection/src/eth/fee-proxy-detector.ts

```diff
@@ -28,24 +33,25 @@ export default class ETHFeeProxyDetector extends FeeReferenceBasedDetector<Payme
   }
 
   /**
-   * Extracts the balance and events of an address
+   * Extracts payment events of an address matching an address and a payment reference
    *
-   * @private
    * @param address Address to check
    * @param eventName Indicate if it is an address for payment or refund
-   * @param network The id of network we want to check
+   * @param requestCurrency The request currency
    * @param paymentReference The reference to identify the payment
-   * @param paymentNetworkVersion the version of the payment network
+   * @param paymentNetwork the payment network
    * @returns The balance
    */
   protected async extractEvents(
     address: string,
     eventName: PaymentTypes.EVENTS_NAMES,
-    network: string,
+    requestCurrency: RequestLogicTypes.ICurrency,
```

Comment:
That's a bit weird, for an ETHFeeProxy request, we know that the request currency is <native-token> no ?